### PR TITLE
Added a note about hardware requirements for compiling on Linux 

### PR DESCRIPTION
**Pull request by art-divin**
When I was setting up a Linux VM environment for debugging Swift compiler crash, I faced a number of issues due to low disk space (`update-checkout` would hang before starting concurrent checkouts) and due to low RAM size (linker would start failing when linking symbols for `bin/clang-17` and other binaries).

This PR adds a note about the minimal RAM requirements, and updates the recommended minimum available disk space.

My findings for when running a Linux VM and compiling the compiler are as follows:

1. 8 GB of RAM with `--release-debuginfo` flag
2. 32 GB of RAM with `--debug` flag
3. 150 GB of disk space

### CODE REVIEW

File: docs/HowToGuides/GettingStarted.md
Comment (finagolfin) [inline docs/HowToGuides/GettingStarted.md:194]: "running Linux in a VM" -> "building on linux" as the same requirements outside VM
"8 GB (when using flag `--release-debuginfo`)" -> "8 GB of RAM when using flag `--release-debuginfo`"
"32 GB (when using flag `--debug-swift`) of RAM and 150 GB" -> "32 GB of RAM when using flag `--debug-swift`, plus 150 GB"

I think you can cut the last two sentences you added, starting with "To speed up."

Comment (art-divin) [reply docs/HowToGuides/GettingStarted.md]: 👋🏻  Hello @finagolfin ,

thank you very much for your input (and posts on Swift Forums!). Really essential, always.

Updated the text 👍🏻

File: docs/HowToGuides/GettingStarted.md
Comment (finagolfin) [inline docs/HowToGuides/GettingStarted.md:194]: While it may be a bit redundant, it reads better if you write "GB of RAM" twice, rather than tuck that away at the end.

**Review by finagolfin — APPROVED** [review]
Other than nit on phrasing, seems worthwhile.

File: docs/HowToGuides/GettingStarted.md
Comment (art-divin) [reply docs/HowToGuides/GettingStarted.md:194]: Thank you, improved! 🔥

File: docs/HowToGuides/GettingStarted.md
Comment (finagolfin) [inline docs/HowToGuides/GettingStarted.md:194]: and 150 -> , plus 150

Too many `and`s otherwise

Comment (art-divin) [reply docs/HowToGuides/GettingStarted.md]: Adjusted 🤝

File: docs/HowToGuides/GettingStarted.md
Comment (AnthonyLatsis) [inline docs/HowToGuides/GettingStarted.md:195]: * I think this should go under "System Requirements". Perhaps we should update the default disk space requirement to 100GB too.
* What’s taking up 150GB on Linux? The entire checkout plus a `--release-debuginfo --debug-swift` build is only about 70GB for me on macOS.
* How much RAM was not enough to pull through a `--debug-swift` build, 16GB?

Comment (art-divin) [reply docs/HowToGuides/GettingStarted.md]: > I think this should go under "System Requirements". Perhaps we should update the default disk space requirement to 100GB too.

Amazing, I did not even notice this section! Surely, updated 🤝 

> What’s taking up 150GB on Linux? The entire checkout plus a --release-debuginfo --debug-swift build is only about 70GB for me on macOS.

--debug-swift took 78 GB of disk space, plus according to [Development Tips](https://github.com/apple/swift/blob/main/docs/DevelopmentTips.md) it is recommended to set sccache `SCCACHE_CACHE_SIZE` environment variable to 50 GB. Then, system size is about 19 GB. 
I've ran out of space mostly because I am still unable to compile with `--debug-swift --debug-foundation` flags, and have to resort to `--release-debuginfo -T --debug-swift-stdlib` to make some progress in crash investigation. Nonetheless, 150 GB is enough to avoid running into a problem with low disk space in the middle of LLVM compilation, for example.

|Folders|Size|
:-:|:-:
image|image

> How much RAM was not enough to pull through a --debug-swift build, 16GB?
Answering your question, 4 GB was my initial VM setup, and I wasted 3 days figuring out why linking was failing, discovered the fact by accident. So my intention with this PR is to let others know how to avoid redundant investigations and work.

Even with 32 GB of RAM, linking of `clang-17` binary fails every single time in a VM. On hardware Linux, it'd be better I presume; however, after 2-3 tries I am able to link largest binaries. On the other hand, if to limit concurrent tasks with `-j 5`, for instance, failures are more rare, but still occur for the same reason, not enough RAM available.

**Review by AnthonyLatsis — APPROVED** [review]
Thank you! Just one more suggestion.

File: docs/HowToGuides/GettingStarted.md
Comment (art-divin) [reply docs/HowToGuides/GettingStarted.md:62]: Very nice rephrasing, thank you for your input 🤝

**Comment by AnthonyLatsis** [comment]
@xedin It’s your call now!

**Review by finagolfin — APPROVED** [review]

**Review by LucianoPAlmeida — APPROVED** [review]

**Comment by art-divin** [comment]
Thank you very much for your approvals 🤝 

I wonder what would be the next step, shall we run CI for such change, or just merge? How to proceed, I'd appreciate any hint 👍🏻

**Comment by AnthonyLatsis** [comment]
We’re waiting for a code owner to also approve your changes.

ping @xedin

**Review by xedin — APPROVED** [review]

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Comment by AnthonyLatsis** [comment]
Thanks!

**Pull request merged**
